cpp/src/copying/concatenate.cu

```diff
@@ -329,58 +330,132 @@ std::unique_ptr<column> concatenate_dispatch::operator()<cudf::struct_view>()
 
 namespace {
 
+void bounds_and_type_check(host_span<column_view const> cols, rmm::cuda_stream_view stream);
+
+/**
+ * @brief Functor for traversing child columns and recursively verifying concatenation
+ * bounds and types.
+ *
+ */
+class traverse_children {
+ public:
+  // nothing to do for simple types.
+  template <typename T>
+  void operator()(host_span<column_view const> cols, rmm::cuda_stream_view stream)
```

Comment:
If we don't touch the parameters, just remove them so we will not have warning "unused parameter....".
```suggestion
  void operator()(host_span<column_view const>, rmm::cuda_stream_view)
```